File: 03_StructuralDesignPatterns/07_ProxyPattern/cpp/badCode.cpp
#include <iostream>
#include <string>
using namespace std;

class RealVideoDownloader {
public:
    string downloadVideo(string videoUrl) {
        // Simulate downloading a video
        cout << "Downloading video from: " << videoUrl << endl;
        return "Video content from " + videoUrl;
    }
};

int main() {
    RealVideoDownloader realVideoDownloader1;
    realVideoDownloader1.downloadVideo("proxy-pattern-video-url");

    RealVideoDownloader realVideoDownloader2;
    realVideoDownloader2.downloadVideo("proxy-pattern-video-url");

    return 0;
}

File: 03_StructuralDesignPatterns/06_ProxyPattern/java/badCode.java
class RealVideoDownloader {
    public String downloadVideo(String videoUrl) {
        // Simulate downloading a video
        System.out.println("Downloading video from: " + videoUrl);
        return "Video content from " + videoUrl;
    }
}

public class badCode {
    public static void main(String[] args) {
        RealVideoDownloader realVideoDownloader1 = new RealVideoDownloader();
        realVideoDownloader1.downloadVideo("proxy-pattern-video-url");

        RealVideoDownloader realVideoDownloader2 = new RealVideoDownloader();
        realVideoDownloader2.downloadVideo("proxy-pattern-video-url");

    }
}